### PR TITLE
Enable and fix use_function_type_syntax_for_parameters lint

### PR DESCRIPTION
**Pull request by kevmoo**
Prepare to release +1 releases for serializable and annotation
Bump SDK constraint on annotation

### CODE REVIEW

File: json_annotation/lib/src/checked_helpers.dart [hunk 1]
```diff
@@ -37,7 +37,7 @@ T $checkedNew<T>(String className, Map map, T constructor(),
 /// `JsonSerializableGenerator.checked` is `true`.
 ///
 /// Should not be used directly.
-T $checkedConvert<T>(Map map, String key, T castFunc(Object value)) {
+T $checkedConvert<T>(Map map, String key, T Function(Object value) castFunc) {
```
Comment (natebosch) [inline json_annotation/lib/src/checked_helpers.dart]: [nit] Consider dropping the argument name. `value` is a noise word that was necessary with the old syntax, but the new syntax does not require it.

File: json_serializable/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 3.2.4+1
+
+- Fix lint affecting `pub.dev` score.
```
Comment (natebosch) [inline json_serializable/CHANGELOG.md]: This should be `3.2.5`.

File: json_serializable/lib/src/type_helper_ctx.dart [hunk 3]
```diff
@@ -62,7 +62,7 @@ class TypeHelperCtx
       (TypeHelper th) => th.deserialize(targetType, expression, this));
 
   Object _run(DartType targetType, String expression,
-          Object invoke(TypeHelper instance)) =>
+          Object Function(TypeHelper instance) invoke) =>
```
Comment (natebosch) [inline json_serializable/lib/src/type_helper_ctx.dart]: [nit] Drop `instance`

File: json_annotation/lib/src/checked_helpers.dart (hunk 1 above)
Comment (kevmoo) [reply json_annotation/lib/src/checked_helpers.dart]: ack

File: json_serializable/lib/src/type_helper_ctx.dart (hunk 3 above)
Comment (kevmoo) [reply json_serializable/lib/src/type_helper_ctx.dart]: ack

**Pull request merged**